delay_distributions.Rmd

```diff
@@ -0,0 +1,206 @@
+
+```r
+df <- hosp_scaled |>
+  pivot_longer(
+    cols = c(infection_date, onset_date, hosp_date),
+    names_to = "type", values_to = "date"
+  )
+ggplot(df, aes(x = date)) +
+  geom_histogram(position = "dodge") +
+  facet_wrap(~ type, ncol = 1) +
+  xlab("Date") +
+  ylab("Count")
+```
+
+# Estimating delay distributions
+
+As mentioned above, our data set of infection, onset and hospitalisation dates is not the typical data set we encounter in outbreaks.
+In reality, we don't ahve infection dates, and we also have to deal with missing data, incomplete observations, data entry errors etc.
+For now, let us just assume we have a data set of symptom onsets and some hospitalisation, and we would like to estimate how long it takes for people to become hospitalised after becoming symptomatic.
+This is an important delay to know about, for example when modelling and forecasting hospitalisations, or more generally for estimating required hospital capacity.
+
+As with all our estimation tasks throughout this course, we will be using stan, a fully-fledged toolkit for estimation.
+We will try to represent the model we simulated above as much as possible in this.
+One way that this could look, for examle, is:
+
+```{r delay_model}
+library("cmdstanr")
+
+delay_model <- '
+  data {
+    int n; // number of data points (rows with hospiatlisation date)
+    array[n] real onset_to_hosp; // onset to hospitalisation delays in the data
+  }
+
+  parameters {
+     real meanlog; // parameter to be estimtated
+     real<lower = 0> sdlog; // parameter to be estimated
+  }
+
+  model {
+    meanlog ~ normal(0, 10); // prior distribution
+    sdlog ~ normal(0, 10) T[0, ]; // prior distribution
+
+    onset_to_hosp ~ lognormal(meanlog, sdlog); // likelihood
+  }
+'
+```
+
+** Take 5 minutes ** to try to understand the model code and the relationships it encodes.
+
+You can now sample from the posterior of this model, i.e. conduct inference, by first compiling the model code.
+
+```{r compile_delay_model}
+file <- write_stan_file(delay_model)
+model <- cmdstan_model(file)
+```
+
+Once this is done, you can _sample_ from the model's posterior distribution:
+
+```{r sample_delay_model}
+model$sample(
+  data = list(
+    n = nrow(na.omit(hosp_scaled)),
+    onset_to_hosp = na.omit(hosp_scaled)$onset_to_hosp
+  )
+```
+
+This will probably cause an error - can you think of a reason why (if you don't see an error you could still think about what may cause an error, or resimulate the data that is being fitted to by re-running the commands that add columns to the infection times data set)?
+
+Can you change something that will avoid the error?
+You can look at [our example](delay_distribution_solution.html#estimating-delay-distributions) for a suggested fix, before moving on to a better solution in the next section.
+
+# Accounting for censoring
+
+A better way to account for censoring is to integrate over a window:
```

Comment:
I thought we discussed this and agree it was best to stick to the solution that is closest to best practice and mirrors the simulation?

Reply:
Would be happy to have a solution with latent infection times but when I did so the stan code turned quite complex. I was probably missing a trick.

Reply:
Suggestion from conversation outside this PR: error on 0-1 delays, which simplifies things